extensions/amp-form/0.1/amp-form.js

```diff
@@ -563,6 +602,10 @@ function updateInvalidTypesClasses(element) {
  * @return {boolean} Whether the element is valid or not.
  */
 function checkUserValidity(element, propagate = false) {
+  // If this is not a field type with checkValidity don't do anything.
```

Comment:
Is the assumption that `amp-selector` somehow will implement this interface? Do we have issues that would track this work? In particular `required` is very sensible for `amp-selector`.